Source/Core/Common/IniFile.cpp

```diff
@@ -250,17 +250,9 @@ bool IniFile::Load(const std::string& filename, bool keep_current_data)
 
   Section* current_section = nullptr;
   bool first_line = true;
-  while (!in.eof())
+  std::string line_str;
+  while (std::getline(in, line_str))
   {
-    std::string line_str;
-    if (!std::getline(in, line_str))
-    {
-      if (in.eof())
-        return true;
-      else
-        return false;
```

Comment:
We should return in.eof() at the end of the function instead of just true in case there was a read error (and get rid of the redundant close).

Reply:
I don't think so. So long as the file can be opened (which is checked by `if (in.fail())` above), `while (std::getline(in, line_str))` will process lines until no more lines can be read, and then we are done, and it was successful. `eof` is not relevant. That's the point of this PR. ;)

Reply:
Well... technically, the read could fail mid-file if you have a disk error, or your USB device gets disconnected, or something like that. In that case you'd want to return `false` and not `true`, since you don't want the caller to proceed with half a file.

Reply:
To handle that situation I think it would be appropriate to just `return in.good();`, which checks for either `failbit` or `badbit`, at the end of the function. I'll update the code.

Reply:
I wanted to put that USB device disconnect idea to the test, so I created a 30GB large Dolphin.ini config file and put it in a symlinked portable user directory on a flashdrive. I'm happy to report that the new code handles this extreme scenario correctly, returning false when ios fails to read in the middle of a file. Granted, the old code also returned the correct thing, but this change also looks nicer.

This change also makes INI loading much faster for an obviously malformed file, because it doesn't need to reallocate the string's storage every loop iteration. So there's that.

Reply:
Are you sure this works correctly? https://en.cppreference.com/w/cpp/io/basic_ios/good claims that `good()` only returns true if none of eofbit/badbit/failbit are set, and the last call to `std::getline()` should set the eofbit...

Reply:
Yeah, I checked it, and this function indeed returns `false` now for the standard 'read to end of file' case. You probably want the https://en.cppreference.com/w/cpp/io/basic_ios/operator_bool instead, which ignores the eofbit. Unfortunately there's no nice function for this so you have to do something like `return in ? true : false;` or `return static_cast<bool>(in);`...

(god i hate iostream)

Reply:
Oh, I completely forgot to test if the function still returns *true* properly. It seems `std::getline` is basically guaranteed to set `eof` *and* `failbit` in this loop's final iteration, because as [cppreference](https://en.cppreference.com/w/cpp/string/basic_string/getline) says...

> If no characters were extracted for whatever reason (not even the discarded delimiter), getline sets [failbit](https://en.cppreference.com/w/cpp/io/ios_base/iostate) and returns.

Dolphin always writes INI files that end with an empty line, so it looks like we're back to returning `in.eof()` here.

Reply:
I've removed the changes from this file.